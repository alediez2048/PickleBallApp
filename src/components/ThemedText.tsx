import React from 'react';
import { Text, TextProps, StyleSheet } from 'react-native';

interface ThemedTextProps extends TextProps {
  variant?: 'body' | 'title' | 'subtitle' | 'caption';
  color?: string;
}

export const ThemedText: React.FC<ThemedTextProps> = ({
  children,
  style,
  variant = 'body',
  color = '#000000',
  ...props
}) => {
  return (
    <Text
      style={[
        { color },
        styles[variant],
        style,
      ]}
      {...props}
    >
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  body: {
    fontSize: 16,
    lineHeight: 24,
  },
  title: {
    fontSize: 24,
    lineHeight: 32,
    fontWeight: 'bold',
  },
  subtitle: {
    fontSize: 18,
    lineHeight: 26,
    fontWeight: '500',
  },
  caption: {
    fontSize: 14,
    lineHeight: 20,
    color: '#666666',
  },
}); 