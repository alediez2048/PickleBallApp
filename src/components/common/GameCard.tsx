import React from 'react';
import { View, Text, StyleSheet, Pressable, Image } from 'react-native';
import type { Game } from '@/types/game';

interface GameCardProps {
  game: Game;
  onPress?: () => void;
}

export const GameCard = React.memo(function GameCard({ 
  game,
  onPress,
}: GameCardProps) {
  return (
    <Pressable
      style={({ pressed }) => [
        styles.container,
        pressed && styles.pressed,
      ]}
      onPress={onPress}
    >
      {game.location.imageUrl && (
        <Image
          source={{ uri: game.location.imageUrl }}
          style={styles.image}
          resizeMode="cover"
        />
      )}
      <View style={styles.content}>
        <Text style={styles.title}>{game.title}</Text>
        <Text style={styles.location}>{game.location.name}</Text>
        <View style={styles.details}>
          <Text style={styles.date}>
            {new Date(game.date).toLocaleDateString()}
          </Text>
          <Text style={styles.players}>
            {game.currentPlayers}/{game.maxPlayers} players
          </Text>
        </View>
        <View style={styles.footer}>
          <Text style={styles.skillLevel}>{game.skillLevel}</Text>
          <Text style={styles.price}>${game.price}</Text>
        </View>
      </View>
    </Pressable>
  );
});

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    borderRadius: 12,
    marginBottom: 16,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  pressed: {
    opacity: 0.7,
  },
  image: {
    width: '100%',
    height: 150,
    borderTopLeftRadius: 12,
    borderTopRightRadius: 12,
  },
  content: {
    padding: 16,
  },
  title: {
    fontSize: 18,
    fontWeight: '600',
    marginBottom: 8,
  },
  location: {
    fontSize: 16,
    color: '#666',
    marginBottom: 8,
  },
  details: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 8,
  },
  date: {
    fontSize: 14,
    color: '#666',
  },
  players: {
    fontSize: 14,
    color: '#666',
  },
  footer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: 8,
  },
  skillLevel: {
    fontSize: 14,
    color: '#666',
    backgroundColor: '#f0f0f0',
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 4,
  },
  price: {
    fontSize: 18,
    fontWeight: '600',
    color: '#2196F3',
  },
}); 