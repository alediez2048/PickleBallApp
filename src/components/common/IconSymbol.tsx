// This file is a fallback for using MaterialIcons on Android and web.

import React from "react";
import { Platform, StyleProp, TextStyle } from "react-native";
import { MaterialIcons } from "@expo/vector-icons";
const { useTheme } = require("@/contexts/ThemeContext");

// Only import SFSymbols on iOS
let SFSymbol: any;
if (Platform.OS === "ios") {
  SFSymbol = require("react-native-sfsymbols").SFSymbol;
}

// Icon name mapping between SF Symbols and Material Icons
const ICON_MAPPING = {
  "house.fill": "home",
  "gamecontroller.fill": "games",
  "person.fill.badge.plus": "person-add",
  "exclamationmark.triangle.fill": "warning",
  "trophy.fill": "emoji-events",
  "person.fill": "person",
  pencil: "edit",
  xmark: "close",
  checkmark: "check",
  "location.fill": "location-on",
  calendar: "calendar-today",
  time: "schedule", // <-- Add Material Icon for 'time'
  "person.2.fill": "people",
  "star.fill": "star",
  "chevron.down": "keyboard-arrow-down",
  "creditcard.fill": "credit-card",
  "gearshape.fill": "settings",
  filter: "filter-list", // <-- Add Material Icon for 'filter'
} as const;

const SF_SYMBOL_MAPPING: Record<IconName, string> = {
  "house.fill": "house.fill",
  "gamecontroller.fill": "gamecontroller.fill",
  "person.fill.badge.plus": "person.fill.badge.plus",
  "exclamationmark.triangle.fill": "exclamationmark.triangle.fill",
  "trophy.fill": "trophy.fill",
  "person.fill": "person.fill",
  pencil: "pencil",
  xmark: "xmark",
  checkmark: "checkmark",
  "location.fill": "location.fill",
  calendar: "calendar",
  time: "clock", // <-- SF Symbol for 'time'
  "person.2.fill": "person.2.fill",
  "star.fill": "star.fill",
  "chevron.down": "chevron.down",
  "creditcard.fill": "creditcard.fill",
  "gearshape.fill": "gearshape.fill",
  filter: "line.3.horizontal.decrease", // <-- SF Symbol for filter
};

type IconName =
  | "person.fill"
  | "house.fill"
  | "gamecontroller.fill"
  | "person.fill.badge.plus"
  | "exclamationmark.triangle.fill"
  | "trophy.fill"
  | "pencil"
  | "xmark"
  | "checkmark"
  | "location.fill"
  | "calendar"
  | "time" // <-- Add 'time' to IconName
  | "person.2.fill"
  | "star.fill"
  | "chevron.down"
  | "creditcard.fill"
  | "gearshape.fill"
  | "filter"; // <-- Add 'filter' to IconName

interface IconSymbolProps {
  name: IconName;
  size?: number;
  color?:
    | "default"
    | "black"
    | "white"
    | "primary"
    | "secondary"
    | "text"
    | string;
  style?: StyleProp<TextStyle>;
}

export function IconSymbol({
  name,
  size = 24,
  color = "default",
  style,
}: IconSymbolProps) {
  // Determine the color to use
  let themedColor = color;
  const { colors } = useTheme();
  // Accept colorType or a valid hex color string
  const isHexColor = (val: string) => /^#([0-9A-Fa-f]{6})$/.test(val);

  if (typeof color === "string" && isHexColor(color)) {
    themedColor = color;
  } else {
    try {
      switch (color) {
        case "black":
          themedColor = colors.black;
          break;
        case "white":
          themedColor = colors.white;
          break;
        case "primary":
          themedColor = colors.primary;
          break;
        case "secondary":
          themedColor = colors.secondary;
          break;
        case "text":
          themedColor = colors.text;
          break;
        default:
          themedColor = colors.text;
      }
    } catch (e) {
      // fallback to provided color
    }
  }

  // Use SF Symbols on iOS, Material Icons elsewhere
  if (Platform.OS === "ios") {
    return (
      <SFSymbol
        name={SF_SYMBOL_MAPPING[name]}
        size={size}
        color={themedColor}
        style={style}
      />
    );
  }

  return (
    <MaterialIcons
      name={ICON_MAPPING[name]}
      size={size}
      color={themedColor}
      style={style}
    />
  );
}
