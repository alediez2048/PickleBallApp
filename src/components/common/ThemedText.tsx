import {
  Text,
  type TextProps,
  StyleSheet,
  type StyleProp,
  type TextStyle,
} from "react-native";
import { useTheme } from "@/contexts/ThemeContext";

export type ThemedTextProps = TextProps & {
  type?:
    | "default"
    | "defaultSemiBold"
    | "title"
    | "subtitle"
    | "subtitleCenter"
    | "paragraph"
    | "paragraphCenter"
    | "caption"
    | "link"
    | "bold"
    | "sectionTitle"
    | "badge"
    | "emptyStateTitle"
    | "emptyStateText"
    | "button"
    | "buttonDisabled"
    | "buttonCancel"
    | "buttonWaitlist";
  colorType?: "default" | "black" | "white" | "primary" | "secondary";
};

export function ThemedText({
  style,
  type = "default",
  colorType = "default",
  ...rest
}: ThemedTextProps) {
  const { colors } = useTheme();

  const getTypeStyle = (): StyleProp<TextStyle> => {
    switch (type) {
      case "title":
        return styles.title;
      case "defaultSemiBold":
        return styles.defaultSemiBold;
      case "subtitle":
        return styles.subtitle;
      case "subtitleCenter":
        return [styles.subtitle, { textAlign: "center" }];
      case "paragraph":
        return styles.paragraph;
      case "paragraphCenter":
        return [styles.paragraph, { textAlign: "center" }];
      case "caption":
        return styles.caption;
      case "link":
        return [styles.link, { color: colors.primary }];
      case "bold":
        return styles.bold;
      case "sectionTitle":
        return {
          fontSize: 18,
          fontWeight: "700",
          color: "#333",
          marginBottom: 4,
        };
      case "badge":
        return { fontSize: 13, fontWeight: "600" };
      case "emptyStateTitle":
        return {
          fontSize: 18,
          fontWeight: "600",
          color: "#000",
          marginBottom: 8,
        };
      case "emptyStateText":
        return {
          fontSize: 14,
          color: "#666",
          textAlign: "center",
          lineHeight: 20,
        };
      case "button":
        return { fontSize: 16, fontWeight: "600", color: "#fff" };
      case "buttonDisabled":
        return { fontSize: 16, fontWeight: "600", color: "#666" };
      case "buttonCancel":
        return { fontSize: 16, fontWeight: "600", color: "#fff" };
      case "buttonWaitlist":
        return { fontSize: 16, fontWeight: "600", color: "#fff" };
      default:
        return styles.default;
    }
  };

  const getColorStyle = (): StyleProp<TextStyle> => {
    switch (colorType) {
      case "black":
        return { color: colors.black };
      case "white":
        return { color: colors.white };
      case "primary":
        return { color: colors.primary };
      case "secondary":
        return { color: colors.secondary };
      default:
        return { color: colors.text };
    }
  };

  const styleMatch = getTypeStyle();
  const colorMatch = getColorStyle();

  return <Text style={[colorMatch, styleMatch, style]} {...rest} />;
}

const styles = StyleSheet.create({
  default: {
    fontSize: 16,
    lineHeight: 24,
  },
  defaultSemiBold: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: "600",
  },
  title: {
    fontSize: 32,
    fontWeight: "bold",
    lineHeight: 36,
  },
  subtitle: {
    fontSize: 20,
    fontWeight: "bold",
    lineHeight: 28,
  },
  paragraph: {
    fontSize: 15,
    lineHeight: 22,
  },
  caption: {
    fontSize: 12,
    color: "#999",
  },
  link: {
    fontSize: 16,
    lineHeight: 24,
    textDecorationLine: "underline",
  },
  bold: {
    fontSize: 16,
    fontWeight: "bold",
  },
});
