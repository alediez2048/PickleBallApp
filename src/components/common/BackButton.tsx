import React from "react";
import { TouchableOpacity, StyleSheet } from "react-native";
import { useRouter } from "expo-router";
import { ThemedText } from "@/components/common/ThemedText";

export default function BackButton() {
  const router = useRouter();

  return (
    <TouchableOpacity style={styles.backButton} onPress={() => router.back()}>
      <ThemedText>‚Üê Back</ThemedText>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  backButton: {
    position: "absolute",
    top: 16,
    left: 16,
    zIndex: 10,
    marginBottom: 16,
  },
});
