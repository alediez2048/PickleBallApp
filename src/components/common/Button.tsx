import React from "react";
import {
  TouchableOpacity,
  StyleSheet,
  View,
  ActivityIndicator,
  TouchableOpacityProps,
  Platform,
} from "react-native";
import { ThemedText } from "@/components/common/ThemedText";

interface ButtonProps extends TouchableOpacityProps {
  variant?: "primary" | "secondary" | "outline" | "danger";
  size?: "small" | "medium" | "large";
  loading?: boolean;
  fullWidth?: boolean;
  children: React.ReactNode;
}

export const Button: React.FC<ButtonProps> = ({
  variant = "primary",
  size = "medium",
  loading = false,
  disabled = false,
  fullWidth = false,
  style,
  children,
  accessibilityLabel,
  accessibilityHint,
  onPress,
  ...props
}) => {
  const isDisabled = disabled || loading;

  const getButtonStyle = () => {
    const variantStyles = {
      primary: styles.primary,
      secondary: styles.secondary,
      outline: styles.outline,
      danger: styles.danger,
    };

    const sizeStyles = {
      small: styles.small,
      medium: styles.medium,
      large: styles.large,
    };

    const computedStyles = [
      styles.base,
      variantStyles[variant],
      sizeStyles[size],
      fullWidth && styles.fullWidth,
      isDisabled && styles.disabled,
      style,
    ];

    // Debug logging
    console.debug("[Button Render]", {
      text: typeof children === "string" ? children : "Custom children",
      variant,
      size,
      isDisabled,
      fullWidth,
      customStyle: style,
      computedStyles: computedStyles
        .filter(
          (s): s is NonNullable<typeof s> => s !== null && s !== undefined
        )
        .map((s) => (typeof s === "object" ? Object.keys(s) : [])),
    });

    return computedStyles;
  };

  const getTextStyle = () => {
    const variantTextStyles = {
      primary: styles.primaryText,
      secondary: styles.secondaryText,
      outline: styles.outlineText,
      danger: styles.dangerText,
    };

    const computedTextStyles = [
      styles.text,
      variantTextStyles[variant],
      isDisabled && styles.disabledText,
    ];

    return computedTextStyles;
  };

  const buttonLabel =
    typeof children === "string" ? children : accessibilityLabel;

  return (
    <TouchableOpacity
      testID='button'
      style={getButtonStyle()}
      disabled={isDisabled}
      onPress={onPress}
      accessibilityRole='button'
      accessibilityLabel={buttonLabel}
      accessibilityHint={accessibilityHint}
      accessibilityState={{
        disabled: isDisabled,
        busy: loading,
      }}
      {...props}
    >
      {loading ? (
        <View style={styles.loadingContainer}>
          <ActivityIndicator
            testID='loading-spinner'
            color={variant === "outline" ? "#4CAF50" : "#FFFFFF"}
            accessibilityLabel='Loading'
          />
        </View>
      ) : typeof children === "string" ? (
        <ThemedText style={getTextStyle()}>{children}</ThemedText>
      ) : (
        children
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  base: {
    borderRadius: 25,
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row",
    borderWidth: 1,
    borderColor: "transparent",
  },
  primary: {
    backgroundColor: "#4CAF50",
    borderColor: "#4CAF50",
  },
  secondary: {
    backgroundColor: "#2196F3",
    borderColor: "#2196F3",
  },
  outline: {
    backgroundColor: "transparent",
    borderColor: "#4CAF50",
  },
  danger: {
    backgroundColor: "#DC2626",
    borderColor: "#DC2626",
  },
  small: {
    paddingVertical: 8,
    paddingHorizontal: 20,
    minHeight: 36,
  },
  medium: {
    paddingVertical: 12,
    paddingHorizontal: 28,
    minHeight: 44,
  },
  large: {
    paddingVertical: 14,
    paddingHorizontal: 32,
    minHeight: 52,
  },
  fullWidth: {
    width: "100%",
  },
  disabled: {
    opacity: 0.5,
    backgroundColor: "#E0E0E0",
    borderColor: "#E0E0E0",
  },
  text: {
    fontSize: 16,
    fontWeight: "600",
    textAlign: "center",
    letterSpacing: 0.5,
  },
  primaryText: {
    color: "#FFFFFF",
  },
  secondaryText: {
    color: "#FFFFFF",
  },
  outlineText: {
    color: "#4CAF50",
  },
  dangerText: {
    color: "#FFFFFF",
  },
  disabledText: {
    color: "#666666",
  },
  loadingContainer: {
    minHeight: 24,
    justifyContent: "center",
    alignItems: "center",
  },
});
