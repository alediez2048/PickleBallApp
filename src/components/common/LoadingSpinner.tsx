import React from "react";
import {
  ActivityIndicator,
  View,
  StyleSheet,
  SafeAreaView,
} from "react-native";
import { ThemedText } from "@/components/common/ThemedText";
import { ThemedView } from "@/components/common/ThemedView";
import { useTheme } from "@/contexts/ThemeContext";
import { LogBox } from "react-native";

interface LoadingSpinnerProps {
  size?: "small" | "large";
  message?: string;
  color?: string;
}

export const LoadingSpinner: React.FC<LoadingSpinnerProps> = ({
  size = "large",
  message,
  color,
}) => {
  const { colors } = useTheme();
  const spinnerColor = colors.text;

  return (
    <ThemedView>
      <SafeAreaView style={styles.container} testID='loading-spinner'>
        <ThemedView style={styles.content}>
          <ActivityIndicator size={size} color={spinnerColor} />
          {message && <ThemedText style={styles.message}>{message}</ThemedText>}
        </ThemedView>
      </SafeAreaView>
    </ThemedView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  content: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  message: {
    marginTop: 12,
    fontSize: 16,
    textAlign: "center",
  },
});

LogBox.ignoreLogs([
  'Warning: Each child in a list should have a unique "key" prop',
]);
