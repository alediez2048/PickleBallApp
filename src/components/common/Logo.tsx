import React from "react";
import { View, StyleSheet } from "react-native";
import { ThemedText } from "@/components/common/ThemedText";

export default function Logo() {
  return (
    <View style={styles.logoContainer}>
      <ThemedText type='title' style={styles.logo}>
        PicklePass
      </ThemedText>
    </View>
  );
}

const styles = StyleSheet.create({
  logoContainer: {
    alignItems: "center",
    marginBottom: 10,
  },
  logo: {
    fontSize: 32,
    fontWeight: "bold",
  },
});
