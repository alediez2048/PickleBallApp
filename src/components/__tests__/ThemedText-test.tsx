import React from "react";
import renderer from "react-test-renderer";
import { ThemedText } from "@/components/common/ThemedText";

// Mock the UIContext
jest.mock('@/contexts/UIContext', () => ({
  UIProvider: ({ children }: { children: React.ReactNode }) => children,
  useUIState: () => ({
    colorScheme: 'light',
    theme: 'default'
  }),
  useThemedColor: () => ({
    text: '#000000',
    background: '#FFFFFF'
  })
}));

describe("ThemedText", () => {
  it("renders with default props", () => {
    const tree = renderer
      .create(<ThemedText>Test text</ThemedText>)
      .toJSON();
    
    expect(tree).toMatchSnapshot('default props');
  });

  it("renders with different types", () => {
    const titleTree = renderer
      .create(<ThemedText type="title">Title text</ThemedText>)
      .toJSON();
    
    const subtitleTree = renderer
      .create(<ThemedText type="subtitle">Subtitle text</ThemedText>)
      .toJSON();
    
    const defaultSemiBoldTree = renderer
      .create(<ThemedText type="defaultSemiBold">Semi Bold text</ThemedText>)
      .toJSON();
    
    expect(titleTree).toMatchSnapshot('title type');
    expect(subtitleTree).toMatchSnapshot('subtitle type');
    expect(defaultSemiBoldTree).toMatchSnapshot('defaultSemiBold type');
  });

  it("renders with custom color", () => {
    const tree = renderer
      .create(<ThemedText style={{ color: "#FF0000" }}>Test text</ThemedText>)
      .toJSON();
    
    expect(tree).toMatchSnapshot('custom color');
  });

  it("renders with custom styles", () => {
    const customStyle = { marginTop: 10, fontSize: 18 };
    
    const tree = renderer
      .create(<ThemedText style={customStyle}>Test text</ThemedText>)
      .toJSON();
    
    expect(tree).toMatchSnapshot('custom styles');
  });

  it("renders with additional props", () => {
    const tree = renderer
      .create(
        <ThemedText 
          numberOfLines={2} 
          ellipsizeMode="tail"
        >
          Test text that might be too long and needs to be truncated
        </ThemedText>
      )
      .toJSON();
    
    expect(tree).toMatchSnapshot('with additional props');
  });
});
