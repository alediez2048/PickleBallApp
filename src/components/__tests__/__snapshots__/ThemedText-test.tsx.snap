// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedText renders with additional props: with additional props 1`] = `null`;

exports[`ThemedText renders with custom color: custom color 1`] = `null`;

exports[`ThemedText renders with custom styles: custom styles 1`] = `null`;

exports[`ThemedText renders with default props: default props 1`] = `null`;

exports[`ThemedText renders with different types: defaultSemiBold type 1`] = `null`;

exports[`ThemedText renders with different types: subtitle type 1`] = `null`;

exports[`ThemedText renders with different types: title type 1`] = `null`;
